bin/convert-mobile.js

```diff
@@ -42,118 +48,171 @@ function convertSingleIconToPng(iconName, width, height, outputRoot, outputSuffi
-  .option('s', {alias: 'outSize', describe: 'size of output image', type: 'string', demandOption: false})
-  .argv;
-
-if (options.outputDir) {
-  outputRoot = path.join(__dirname, options.outputDir);
+/**
+ * Indexes all calcite icons contained in directory at path
+ * @param {string} baseIconPath - path to calcite .svg icons directory
+ */
+async function indexCalciteIcons(baseIconPath) {
+  return new Promise(resolve => {
+    var iconIndex = {};
+    fs.readdir(baseIconPath, function (error, files) {
+      if (error) {
+        console.log(error);
+        process.exit(1);
+      }
+      files.forEach(function (file) {
+        // strip all files of file size information, catalog in an index
+        var base_name = path.basename(file);
+        base_name = base_name.replace('.svg', '');
+        var size = undefined;
+        if (base_name.includes('-16')) {
+          base_name = base_name.replace('-16', '');
+          size = 16;
+        }
+        else if (base_name.includes('-24')) {
+          base_name = base_name.replace('-24', '');
+          size = 24;
+        }
+        else if (base_name.includes('-32')) {
+          base_name = base_name.replace('-32', '');
+          size = 32;
+        }
+        if (!iconIndex[base_name]) {
+          iconIndex[base_name] = {}
+        }
+        iconIndex[base_name][size] = path.join(baseIconPath, file);
+      });
+      resolve(iconIndex);
+    });
+  });
 }
 
-// update output size if set
-if (options.outSize){
-  outputSize = parseInt(options.outSize);
+/**
+ * Indexes all calcite icons contained in directory at path
+ * @param {string} xcAssetsBaseDirectory - path where to derive calcite.xcassets
+ */
+async function createCalciteXCAssets(xcAssetsBaseDirectory) {
+  return new Promise(resolve => {
+    // Put in .xcassets folder
+    var directory = path.join(xcAssetsBaseDirectory, 'calcite.xcassets');
+    // Make sure dir exists
+    if (!fs.existsSync(directory)) {
+      fs.mkdirSync(directory, {
+        recursive: true
+      });
+    }
+    // read contents.json template
+    let template_path = path.join(__dirname, 'templates', 'xcassets.json');
+    // write out file
+    fs.readFile(template_path, 'utf8', function (error, buffer) {
+      if (error) {
+        console.log(error);
+        process.exit(1);
+      }
+      const contents_output_path = path.join(directory, 'Contents.json');
+      fs.writeFile(contents_output_path, buffer, function (error) {
+        resolve(directory);
+      });
+    });
+  });
 }
 
-// update source icon size based on output size if source not explicitly set
-if (!options.inSize && options.outSize){
-  if (options.outSize < 24){
-    sourceIconSuffix = '-16.svg';
-  } else if (options.outSize < 32){
-    sourceIconSuffix = '-24.svg';
-  } else {
-    sourceIconSuffix = '-32.svg';
+async function main() {
+  // index all calcite icons
+  let iconIndex = await indexCalciteIcons('./icons/');
```

Comment:
`indexCalciteIcons` is generating a structure very similar to that of `docs/icons.json`. Almost wondering if you could just import that file and use it directly.

Reply:
Oh wow, I didn't know this file existed. Is it the complete set including the .svg vectors?

Reply:
Yeah it's generated from reading `icons/*.svg` I believe. Here's the code that generates it: https://github.com/Esri/calcite-ui-icons/blob/master/bin/path-data.js#L85-L130

Reply:
I suppose either this file or the directory serve as source of truth, is one better than the other? If it's all the same to you... can I keep as-is?

Reply:
fine by me 🌈 🌟